internal/controller/components/kueue/kueue_support.go

```diff
@@ -46,3 +62,145 @@ func kueueConfigManifestsPath() odhtypes.ManifestInfo {
 		SourcePath: "",
 	}
 }
+
+func getManagedNamespaces(ctx context.Context, c client.Client) ([]corev1.Namespace, error) {
+	// Deduped namespaces, since some might have both labels.
+	var uniqueNamespaces = make(map[string]corev1.Namespace)
+
+	// Add all namespaces with management label.
+	if err := collectNamespacesWithPagination(ctx, c, uniqueNamespaces, client.MatchingLabels{
+		KueueManagedAnnotationKey: "true",
+	}); err != nil {
+		return nil, err
+	}
+	// Add namespaces with legacy management label.
+	if err := collectNamespacesWithPagination(ctx, c, uniqueNamespaces, client.MatchingLabels{
+		KueueLegacyManagedAnnotationKey: "true",
+	}); err != nil {
+		return nil, err
+	}
+
+	managedNamespacesList := []corev1.Namespace{}
+	for v := range maps.Values(uniqueNamespaces) {
+		managedNamespacesList = append(managedNamespacesList, v)
+	}
+
+	return managedNamespacesList, nil
+}
+
+func collectNamespacesWithPagination(ctx context.Context, c client.Client, namespaceSet map[string]corev1.Namespace, opts ...client.ListOption) error {
+	lo := client.ListOptions{
+		Limit: NSListLimit,
+	}
+	opts = append(opts, &lo)
+	for {
+		// Listing namespaces with management label
+		namespaces := &corev1.NamespaceList{}
+		if err := c.List(ctx, namespaces, opts...); err != nil {
+			return fmt.Errorf("failed to list namespaces with label %s: %w", KueueManagedAnnotationKey+"=true", err)
+		}
+
+		for _, ns := range namespaces.Items {
+			namespaceSet[ns.Name] = ns
+		}
+
+		if namespaces.Continue == "" {
+			break
+		}
+
+		lo.Continue = namespaces.Continue
+	}
+	return nil
+}
+
+// i.e. if a namespace has just the KueueLegacyManagedAnnotationKey or the KueueManagedAnnotationKey, the other one is added as well.
+func fixAnnotationsOfManagedNamespaces(ctx context.Context, c client.Client, namespaces []corev1.Namespace) error {
+	for _, ns := range namespaces {
+		_, legacyManagedLabelPresent := ns.Labels[KueueLegacyManagedAnnotationKey]
+		_, managedLabelPresent := ns.Labels[KueueManagedAnnotationKey]
+		if legacyManagedLabelPresent && !managedLabelPresent {
+			ns.Labels[KueueManagedAnnotationKey] = "true"
+			if err := c.Update(ctx, &ns); err != nil {
+				return err
+			}
+		}
+		if !legacyManagedLabelPresent && managedLabelPresent {
+			ns.Labels[KueueLegacyManagedAnnotationKey] = "true"
+			if err := c.Update(ctx, &ns); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
```

Comment:
I'd suggest to use` resources.HasLabel/SetLabels `as it handles the case where `Labels` is `nil`, something like:

```suggestion
func fixAnnotationsOfManagedNamespaces(ctx context.Context, c client.Client, namespaces []corev1.Namespace) error {
        for _, ns := range namespaces {
		hasLegacy := resources.HasLabel(&ns, KueueLegacyManagedAnnotationKey)
		hasManaged := resources.HasLabel(&ns, KueueManagedAnnotationKey)
		
		// Skip if both labels are already present
		if hasLegacy && hasManaged {
			continue
		}
		
		// Set both labels to ensure consistency
		resources.SetLabels(&ns, map[string]string {
		    KueueLegacyManagedAnnotationKey: "true".
		    KueueManagedAnnotationKey: "true",
		})
		
		if err := c.Update(ctx, &ns); err != nil {
			return client.IgnoreNotFound(err)
		}
	}
	return nil
}
```